src/os/bluestore/bluefs_types.h

```diff
@@ -330,6 +330,7 @@ struct bluefs_transaction_t {
     op_bl.claim_append(from.op_bl);
   }
 
+  DENC_HELPERS
```

Comment:
i think the compiler warning is a signal that we are not using the `DENC` macro for defining the dencoders of `bluefs_transaction_t`. in fact, `bluefs_transaction_t` defines its own dencoders manually without relying on `DENC`. what `DENC_HELPERS` provides to this class is but 
```c++
   static void _denc_start(size_t& p,                                    \
                           __u8 *struct_v,                               \
                           __u8 *struct_compat,                          \
                           char **, uint32_t *) {                        \
     p += 2 + 4;                                                         \
   }                                                                     \
```
for calculating the size of the header of the serialized `bluefs_transaction_t`. and `DENC_FINISH()` here is a noop.

so i'd suggest apply following patch
```diff
diff --git a/src/os/bluestore/bluefs_types.cc b/src/os/bluestore/bluefs_types.cc
index ff9c96ab660..a17c3bc8a3c 100644
--- a/src/os/bluestore/bluefs_types.cc
+++ b/src/os/bluestore/bluefs_types.cc
@@ -228,10 +228,10 @@ std::ostream& operator<<(std::ostream& out, const bluefs_fnode_delta_t& delta)

 // bluefs_transaction_t

-DENC_HELPERS
 void bluefs_transaction_t::bound_encode(size_t &s) const {
   uint32_t crc = op_bl.crc32c(-1);
-  DENC_START(1, 1, s);
+  // struct_v, struct_compat, and len
+  s += 2 + 4;
   denc(uuid, s);
   denc_varint(seq, s);
   // not using bufferlist encode method, as it merely copies the bufferptr and not
@@ -242,7 +242,6 @@ void bluefs_transaction_t::bound_encode(size_t &s) const {
     s += it.length();
   }
   denc(crc, s);
-  DENC_FINISH(s);
 }

 void bluefs_transaction_t::encode(bufferlist& bl) const
```

instead of using `DENC_HELPERS` for a single line code.

Reply:
@cbodley hi Casey, i am adding you as a reviewer. as you also tried to address this issue in #54816

Reply:
maybe the macro should qualify the helper functions with `[[maybe_unused]]`?